src/hammer-vlsi/synthesis/vivado/__init__.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List, Any
+
+import os
+import errno
+import pkg_resources
+from itertools import chain
+
+from jinja2 import Template
+
+from hammer_vlsi import HammerSynthesisTool, HammerToolStep, deepdict
+
+_temp_dir = 'file_templates'
+_synth_tag = 'vivado_synth'
+
+class VivadoSynth(HammerSynthesisTool):
+
+
+    _base_files = ['board.tcl', 'paths.tcl', 'project.tcl']
+    
+    @property
+    def env_vars(self) -> Dict[str, str]:
+        new_dict = deepdict(super().env_vars)
+        return new_dict
+
+    @property
+    def steps(self) -> List[HammerToolStep]:
+        return self.make_steps_from_methods([
+            self.run_synthesis,
+            self.run_place_and_route,
+        ])
+
+    def _get_dcp_path(self):
```

Comment:
We can just use `def get_dcp_path` here methinks. Also in general we've been trying to add type annotations for all functions, including return types.